doc/specs/#888 - Com Api.md

```diff
@@ -0,0 +1,889 @@
+        PackageMatchFilter MatchCriteria { get; };
+    }
+    /// IMPLEMENTATION NOTE: SearchResult from AppInstallerRepositorySearch.h
+    /// Search result data returned from FindPackages
+        [contract(Microsoft.Management.Deployment.WindowsPackageManagerContract, 1)]
+    runtimeclass FindPackagesResult
+    {
+        /// The full set of results from the search.
+        Windows.Foundation.Collections.IVectorView<ResultMatch> Matches { get; };
+
+        /// If true, the results were truncated by the given SearchRequest::MaximumResults.
+        /// USAGE NOTE: Windows Package Manager does not support result pagination, there is no way to continue 
+        /// getting more results.
+        Boolean IsTruncated{ get; };
+    }
+
+    /// Options for FindPackages
+    [contract(Microsoft.Management.Deployment.WindowsPackageManagerContract, 1)]
+    runtimeclass FindPackagesOptions
+    {
+        FindPackagesOptions();
+
+        /// DESIGN NOTE: 
+        /// This class maps to SearchRequest from  AppInstallerRepositorySearch.h 
+        /// That class is a container for data used to filter the available manifests in an app catalog.
+        /// Its properties can be thought of as:
+        /// (Query || Inclusions...) && Filters...
+        /// If Query and Inclusions are both empty, the starting data set will be the entire database.
+        /// Everything && Filters...
+        /// That has been translated in this api so that 
+        /// Query is PackageMatchField::AppCatalogDefined and PackageMatchFilter.IsAdditive = true
+        /// Inclusions are PackageMatchFilter.IsAdditive = true
+        /// Filters are PackageMatchFilter.IsAdditive = false
+
+        /// Filters to find packages
+        /// USAGE NOTE: Only one filter with PackageMatchField::AppCatalogDefined is allowed.
+        Windows.Foundation.Collections.IVector<PackageMatchFilter> Filters { get; };
+
+        /// Restricts the length of the returned results to the specified count.
+        UInt32 ResultLimit;
+
+        /// Overrides the default search behavior for this search if the catalog is a composite catalog.
+        /// IMPLEMENTATION NOTE: Windows Package Manager currently only supports setting the search behavior when 
+        /// creating the source so this is new work but is expected to be very small.
+        CompositeSearchBehavior CompositeSearchBehavior;
+    }
+
+    /// IMPLEMENTATION NOTE: ISource from AppInstallerRepositorySource.h
+    /// A catalog for searching for packages
+    [contract(Microsoft.Management.Deployment.WindowsPackageManagerContract, 1)]
+    runtimeclass AppCatalog
+    {
+        /// Gets a value indicating whether this app catalog is a composite of other app catalogs,
+        /// and thus the packages may come from disparate app catalogs as well.
+        Boolean IsComposite { get; };
+        /// The details of the app catalog if it is not a composite.

```

Comment:
What happens if you ask for this property when the catalog is composite?

Reply:
? Did this mean if it's not composite? It's false if it's not composite.

Reply:
I mean the `Info` property. Does it return the info for the first catalog in the composite?

Reply:
I was expecting null.

Reply:
DISCUSSION: If it's a composite, there's no easy way to construct a valid `Info` for it, so this is null.

CONSIDER: See if you can construct a dummy `AppCatalogInfo` for composites so at least it's always dottable-through.